Library/Homebrew/extend/pathname.rb

```diff
@@ -466,6 +466,31 @@ def dylib?
   def rpaths
     []
   end
+
+  sig { returns(String) }
+  def magic_number
+    @magic_number ||= if directory?
+      ""
+    else
+      # Length of the longest regex (currently Tar).
+      # The `T.let` is a workaround until we have https://github.com/sorbet/sorbet/pull/6865
+      T.let(binread(262), T.nilable(String)) || ""
```

Comment:
262 was formerly `MAX_MAGIC_NUMBER_LENGTH`, but i didn't want to add more than necessary to the `Pathname` space, or add any redirection.